File: frontend/src/services/imageService.ts
const TOTAL_IMAGES = 15;

export const getProductImage = (productId: number): string => {
  // Usar el ID del producto para obtener una imagen consistente
  const imageNumber = (productId % TOTAL_IMAGES) + 1;
  return `/images/products/${imageNumber}.png`;
};

export const getCartItemImage = (itemId: number, productId?: number): string => {
  if (productId) {
    return getProductImage(productId);
  }
  // Si no hay product_id, usar el itemId para mantener consistencia
  const imageNumber = (itemId % TOTAL_IMAGES) + 1;
  return `/images/products/${imageNumber}.png`;
};

export const getRandomImageForNewProduct = (): string => {
  const randomNumber = Math.floor(Math.random() * TOTAL_IMAGES) + 1;
  return `/images/products/${randomNumber}.png`;
}; 

File: server/server/services/image_service.py
import random

class ImageService:
    TOTAL_IMAGES = 15
    
    @staticmethod
    def get_random_product_image() -> str:
        image_number = random.randint(1, ImageService.TOTAL_IMAGES)
        return f"/images/products/{image_number}.png"
    
    @staticmethod
    def get_image_for_product(product_id: int) -> str:
        # Usar el ID del producto para obtener una imagen consistente
        image_number = (product_id % ImageService.TOTAL_IMAGES) + 1
        return f"/images/products/{image_number}.png" 